samples/SkiaMultiplatformSample/src/commonMain/kotlin/org/jetbrains/skiko/sample/Clocks.kt

```diff
@@ -7,45 +7,60 @@ import org.jetbrains.skia.paragraph.ParagraphStyle
 import org.jetbrains.skia.paragraph.TextStyle
 import org.jetbrains.skiko.SkikoView
 import org.jetbrains.skiko.SkikoPointerEvent
+import org.jetbrains.skiko.SkikoGestureEvent
+import org.jetbrains.skiko.SkikoGestureEventKind
+import org.jetbrains.skiko.SkikoGestureEventState
 import org.jetbrains.skiko.isLeftClick
 import kotlin.math.cos
 import kotlin.math.sin
 import kotlin.math.PI
 
 class Clocks: SkikoView {
     private var frame = 0
-    private var xpos = 0.0
-    private var ypos = 0.0
+    private var xpos = 0.0f
+    private var ypos = 0.0f
+    private var xOffset = 0.0f
+    private var yOffset = 0.0f
+    private var scale = 1.0f
+    private var rotate = 0.0f
     private val fontCollection = FontCollection()
         .setDefaultFontManager(FontMgr.default)
 
-    override fun onRender(canvas: Canvas, width: Int, height: Int, nanoTime: Long) {
+    override fun onRender(canvas: Canvas, w: Int, h: Int, nanoTime: Long) {
+        val xOffset = xOffset
+        val yOffset = yOffset
+        val xpos = xpos * scale + xOffset * scale
+        val ypos = ypos * scale + yOffset * scale
+        val width = (w * scale).toInt() + xOffset.toInt()
+        val height = (h * scale).toInt() + yOffset.toInt()
         val watchFill = Paint().apply { color = 0xFFFFFFFF.toInt() }
         val watchStroke = Paint().apply {
                color = 0xFF000000.toInt()
                mode = PaintMode.STROKE
-               strokeWidth = 1f
+               strokeWidth = 1f * scale
         }
         val watchStrokeAA = Paint().apply {
           color = 0xFF000000.toInt()
           mode = PaintMode.STROKE
-          strokeWidth = 1f
+          strokeWidth = 1f * scale
         }
         val watchFillHover = Paint().apply { color = 0xFFE4FF01.toInt() }
-        for (x in 0 .. (width - 50) step 50) {
-            for (y in 20 .. (height - 50) step 50) {
-                val hover = xpos > x + 0 && xpos < x + 50 && ypos > y + 0 && ypos < y + 50
+        val clockSize = 40f * scale
```

Comment:
Why scale on that level? Cannot we use canvas scaling instead? This code is somewhat error prone.

Reply:
Done.